app/controllers/add-preprint.js

```diff
@@ -95,55 +95,61 @@ export default Ember.Controller.extend(Validations, NodeActionsMixin, {
-            if (path.children && this.get(`filter.${i + 1}.value`)) {
-                return {
-                    name: path.name,
-                    children: path.children.filter(child =>
-                        this.get(`filter.${i + 1}.value`).indexOf(child.name || child) !== -1)
-                };
-            }
-            return path;
-        });
+    * Subjects section
+    */
+    topFilter: '',
+    midFilter: '',
+    botFilter: '',
+    updateFilteredPath() {
+        var _this = this;
+        var overallPath = [];
+        var paths = this.get('path').slice(0, 2);
+        if (paths.length === 1) {
+            _this.get('store').query('taxonomy', { filter: { parent_ids: paths[0].id }, page: { size: 100 } }).then(results => {
+                Ember.set(paths[0], 'children', results.map(
+                    function(result) { return { name: result.get('text'), id: result.id }; }
+                ));
+                overallPath.push(paths[0]);
+                _this.set('filteredPath', overallPath);
+            });
+        } else if (paths.length === 2) {
+            _this.get('store').query('taxonomy', { filter: { parent_ids: paths[0].id }, page: { size: 100 } }).then(results => {
+                Ember.set(paths[0], 'children', results.map(
+                    function(result) { return { name: result.get('text'), id: result.id }; }
+                ));
+                overallPath.push(paths[0]);
+                _this.get('store').query('taxonomy', { filter: { parent_ids: paths[1].id }, page: { size: 100 } }).then(results => {
+                    Ember.set(paths[1], 'children', results.map(
+                        function(result) { return { name: result.get('text'), id: result.id }; }
+                    ));
+                    overallPath.push(paths[1]);
+                    _this.set('filteredPath', overallPath);
+                });
+            });
+        }
+    },
+    filteredPath: Ember.computed('path', function() {
+        this.updateFilteredPath();
     }),
-    sortedTaxonomies: Ember.computed('taxonomies', 'filter', 'filter.0.value', function() {
-        // Format of data that is expected in the hbs
-        return [{
-            name: 'a',
-            children: [{
-                name: 'b',
-                children: ['c', 'd', 'e']
-            }, {
-                name: 'f',
-                children: ['g']
-            }],
-        }, {
-            name: 'h',
-            children: [{
-                name: 'i',
-                children: ['j', 'k']
-            }]
-        }, {
-            name: 'l'
-        }].filter(taxonomy =>
-            !this.get('filter.0.value') || taxonomy.name.indexOf(this.get('filter.0.value')) !== -1
-        );
+    sortedTaxonomies: Ember.computed('taxonomies', function() {
+        var _this = this;
+        this.get('store').query('taxonomy', { filter: { parent_ids: 'null' }, page: { size: 100 } }).then(results => {
```

Comment:
We're probably going to have to merge this as is- but for future reference, I'd encourage you to abstract away common operations like this query. In particular, fetching 100 results can be very slow, and it would be good to define the page size for all these operations in one place in case we decide to change/ edit it later.